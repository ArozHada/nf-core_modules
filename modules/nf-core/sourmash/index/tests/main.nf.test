nextflow_process {

    name "Test Process SOURMASH_INDEX"
    script "../main.nf"
    process "SOURMASH_INDEX"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "sourmash"
    tag "sourmash/index"
    tag "sourmash/sketch"

    setup {
        run("SOURMASH_SKETCH") {
            script "../../sketch/main.nf"
            process {
            """
            input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
            ]
            """
            }
        }
    }

    test("sarscov2 genome [fasta]") {

        when {
            process {
                """
                input[0] = SOURMASH_SKETCH.out.signatures
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() },
                { assert process.out.signature_index }
            )
        }

    }

    test("sarscov2 genome [fasta] - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = SOURMASH_SKETCH.out.signatures
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.signature_index.collect { file(it[1]).getName() } +
                                process.out.versions).match() }
            )
        }

    }

}
