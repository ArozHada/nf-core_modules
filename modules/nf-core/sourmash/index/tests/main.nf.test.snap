{
    "sarscov2 genome [fasta] - stub": {
        "content": [
            [
                "test.sbt.zip",
                "versions.yml:md5,a0f7edb1cdc7f616798c76c05f691c4e"
            ]
        ],
        "timestamp": "2023-12-28T18:42:58.925226193"
    },
    "sarscov2 genome [fasta]": {
        "content": [
            [
                "versions.yml:md5,a0f7edb1cdc7f616798c76c05f691c4e"
            ]
        ],
        "timestamp": "2023-12-28T18:42:42.082444403"
    }
}